web/src/components/dashboard/SidebarCardFormationNudge.test.tsx

```diff
@@ -0,0 +1,110 @@
+import { SidebarCardFormationNudge } from "@/components/dashboard/SidebarCardFormationNudge";
+import { getMergedConfig } from "@/contexts/configContext";
+import { SidebarCardContent } from "@/lib/types/types";
+import {
+  generatePreferences,
+  generateProfileData,
+  generateSidebarCardContent,
+  generateUserData,
+} from "@/test/factories";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import {
+  currentUserData,
+  setupStatefulUserDataContext,
+  userDataWasNotUpdated,
+  WithStatefulUserData,
+} from "@/test/mock/withStatefulUserData";
+import { formationTaskId } from "@businessnjgovnavigator/shared/";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("next/router", () => ({
+  useRouter: jest.fn(),
+}));
+
+const Config = getMergedConfig();
+
+describe("<SidebarCardFormationNudge />", () => {
+  let card: SidebarCardContent;
+
+  const renderWithUserData = (userData: Partial<UserData>) => {
+    render(
+      <WithStatefulUserData initialUserData={generateUserData(userData)}>
+        <SidebarCardFormationNudge card={card} />
+      </WithStatefulUserData>
+    );
+  };
+
+  describe("when clicking funding button", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+      useMockRouter({});
+      setupStatefulUserDataContext();
+      card = generateSidebarCardContent({ id: "formation-nudge" });
+    });
+
+    it("allows user in NEEDS_TO_FORM Phase to see modal", () => {
```

Comment:
this test name isn't quite accurate because there's nothing in the content of the test that indicates anything about "NEEDS_TO_FORM"

a better test name might indicate that the modal becomes visible when the user clicks the CTA button